test/node/bson_test.js

```diff
@@ -2268,4 +2269,117 @@ describe('BSON', function () {
     expect(() => BSON.serialize(badArray)).to.throw();
     expect(() => BSON.serialize(badMap)).to.throw();
   });
+
+  describe('Should support util.inspect for', function () {
+    /**
+     * @ignore
+     */
+    it('Binary', function (done) {
+      const binary = new Binary(Buffer.from('0123456789abcdef0123456789abcdef', 'hex'), 4);
+      expect(inspect(binary)).to.equal('Binary("0123456789abcdef0123456789abcdef", 4)');
+      done();
+    });
+
+    /**
+     * @ignore
+     */
+    it('Code', function (done) {
+      const code = new Code('this.a > i', { i: 1 });
+      expect(inspect(code)).to.equal('Code("this.a > i", {"i":1})');
+      done();
+    });
+
+    /**
+     * @ignore
+     */
+    it('DBRef', function (done) {
+      const oid = new ObjectId('deadbeefdeadbeefdeadbeef');
+      const dbref = new DBRef('namespace', oid, 'integration_tests_');
+      expect(inspect(dbref)).to.equal(
+        'DBRef("namespace", "deadbeefdeadbeefdeadbeef", "integration_tests_")'
+      );
+      done();
+    });
+
+    /**
+     * @ignore
+     */
+    it('Decimal128', function (done) {
+      const dec = Decimal128.fromString('1.42');
+      expect(inspect(dec)).to.equal('Decimal128("1.42")');
+      done();
+    });
+
+    /**
+     * @ignore
+     */
+    it('Int32', function (done) {
+      const int = new Int32(42);
+      expect(inspect(int)).to.equal('Int32(42)');
+      done();
+    });
+
+    /**
+     * @ignore
+     */
+    it('Int32', function (done) {
+      const int = new Int32(42);
+      expect(inspect(int)).to.equal('Int32(42)');
+      done();
+    });
+
+    /**
+     * @ignore
+     */
+    it('Int32', function (done) {
+      const int = new Int32(42);
+      expect(inspect(int)).to.equal('Int32(42)');
+      done();
+    });
```

Comment:
I don't see a difference between these tests. Accidental dupe? 
I think one of these should be `ObjectId` and another should be `Double` perhaps?

Reply:
Oh god 🙈 These were just dupes. I've added an implementation for `Double` nonetheless. The implementation of `ObjectId` currently is in another PR - #412 - and I skipped that here to avoid conflicts. If you'd like to I could move that in and we could close #412.

Reply:
#412 is already merged so looks like we're good to go, thanks for adding Double!